### PR TITLE
[BIT] 引擎 accuracy 回测修复 

### PR DESCRIPTION
**Pull request by Cutelemon6**
# 0523 回测前
回测 amp api config
fused_layer_norm
fused_feedward

# 0523 回测后
### paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm

skip 配置为确定性配置，之前测试发现有 accuracy error，移入 accuracy_error_1.txt，在 base.py 中将该 api 注释。另有原来在 random_calculation.txt 的配置为确定性配置，有一个 amp 测试通过，移入 accuracy_amp_1.txt，其余三个配置测试通过，移入 accuracy_8.txt

### paddle.incubate.nn.functional.fused_feedforward

属于 rand apis，已合入的 PR [#175 move apis ...](https://github.com/PFCCLab/PaddleAPITest/pull/175) 部分修复，从 accuracy_error_1.txt 中移入 2_paddle_only/random_calulation.txt，移除了 fresh_report/accuracy 对应的 test log。另外原来在 random_calulation.txt 中有一部分是确定性配置，测试通过后移入 accuracy_8.txt。

### paddle.incubate.nn.functional.fused_layer_norm

报错指示需要 amp test，在本 PR [#177](https://github.com/PFCCLab/PaddleAPITest/pull/177) 0523 回测前均已测试并移入 accuracy_amp_error_1.txt，从 pending 中删除。

### paddle.Tensor.greater_equal

与 paddle.greater_equal 类似，写入 mapping.json 后测试通过，移入 accuracy_8.txt

### CODE REVIEW

File: report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log [hunk 1]
```diff
@@ -675,3 +675,106 @@ Not equal to tolerance rtol=0.01, atol=0.01
        [-0.2776 ,  0.4312 ,  0.4165 , ..., -0.444  , -0.3582 ,  0.00583],
        [-0.3018 ,  0.02596, -0.0886 , ...,  0.2703 ,  0.391  , -0.4482 ],...
 
+paddle.incubate.nn.functional.fused_layer_norm(Tensor([101, 64],"float16"), Tensor([64],"float32"), Tensor([64],"float32"), 1e-05, begin_norm_axis=1, )
+[accuracy error] paddle.incubate.nn.functional.fused_layer_norm(Tensor([101, 64],"float16"), Tensor([64],"float32"), Tensor([64],"float32"), 1e-05, begin_norm_axis=1, ) 
+ (PreconditionNotMet) Tensor not initialized yet when DenseTensor::place() is called.
+  [Hint: holder_ should not be null.] (at ../paddle/phi/core/dense_tensor_impl.cc:62)
```
Comment (cangtianhuang) [inline report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log:681]: PreconditionNotMet 感觉像是 paddle error，为什么这处报错 accuracy error？

File: tester/api_config/5_accuracy/accuracy_8.txt [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+paddle.incubate.nn.functional.fused_feedforward(x=Tensor([1, 2, 2],"float32"), linear1_weight=Tensor([2, 4],"float32"), linear2_weight=Tensor([4, 2],"float32"), linear1_bias=Tensor([4],"float32"), linear2_bias=Tensor([2],"float32"), ln2_scale=Tensor([2],"float32"), ln2_bias=Tensor([2],"float32"), dropout1_rate=0, dropout2_rate=0, )
+paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm(Tensor([8, 128, 1024],"float32"), Tensor([8, 128, 1024],"float32"), Tensor([1024],"float32"), Tensor([1024],"float32"), Tensor([1024],"float32"), 0.0, 1e-05, )
+paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm(x=Tensor([1, 2, 4],"float32"), residual=Tensor([1, 2, 4],"float32"), bias=None, ln_scale=Tensor([4],"float32"), ln_bias=None, dropout_rate=0.0, ln_epsilon=1e-05, training=True, mode="upscale_in_train", name=None, )
+paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm(x=Tensor([1, 2, 4],"float32"), residual=Tensor([1, 2, 4],"float32"), bias=Tensor([4],"float32"), ln_scale=Tensor([4],"float32"), ln_bias=Tensor([4],"float32"), dropout_rate=0.0, ln_epsilon=1e-05, training=True, mode="upscale_in_train", name=None, )
```
Comment (cangtianhuang) [inline tester/api_config/5_accuracy/accuracy_8.txt:16]: 即使这些配置是确定的，测试时是否仍会跳过它们？

File: tester/paddle_to_torch/rules.py [hunk 3]
```diff
@@ -2079,6 +2079,13 @@ def fused_feedforward(x, linear1_weight, linear2_weight, linear1_bias=None, line
 """
         core = """
 result = fused_feedforward(x, linear1_weight, linear2_weight, linear1_bias, linear2_bias, ln1_scale, ln1_bias, ln2_scale, ln2_bias, dropout1_rate, dropout2_rate, activation, ln1_epsilon, ln2_epsilon, pre_layer_norm, training, mode)
+
+# Force tensors to float16 when autocast is enabled, as all our test cases using autocast expect fp16
+# https://docs.pytorch.org/docs/stable/amp.html#autocast-op-reference
+# Note: Autocast detection must happen inside the core execution block because preprocess and postprocess
+# do not use autocast context manager
+if torch.is_autocast_enabled():
+    result = result.to(torch.float16)
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2088]: core=[core] 改为 core=core.splitlines()

**Review by cangtianhuang — CHANGES_REQUESTED** [review]

File: tester/api_config/5_accuracy/accuracy_8.txt (hunk 2 above)
Comment (Cutelemon6) [reply tester/api_config/5_accuracy/accuracy_8.txt:16]: 不会了，skip 列表中已注释，详见 base.py 中的 TODO

File: report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log (hunk 1 above)
Comment (Cutelemon6) [reply report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log:681]: 猜测是 paddle api 能正常运行并得到 tensor，但是在精度对比时引用该 tensor 会抛出错误

File: tester/api_config/5_accuracy/accuracy_8.txt (hunk 2 above)
Comment (cangtianhuang) [reply tester/api_config/5_accuracy/accuracy_8.txt:16]: 如果要移除 rand api 列表，请一同修复 base.py/need_skip() 方法，其中应用了 paddle_only 测试时跳过 rand api 列表

File: report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log (hunk 1 above)
Comment (cangtianhuang) [reply report/fresh_report/accuracy/test_log_paddle.incubate.nn.functional.fused_layer_norm.log:681]: ？好家伙，这个确实需要上报

File: tester/paddle_to_torch/rules.py (hunk 3 above)
Comment (Cutelemon6) [reply tester/paddle_to_torch/rules.py:2088]: 已修改

**Review by cangtianhuang — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
